docs/reference/glossary.asciidoc

```diff
@@ -61,6 +61,16 @@
   `object`. The mapping also allows you to define (amongst other things)
   how the value for a field should be analyzed.
 
+[[glossary-filter]] filter ::
+
+  A filter is a query. It is a kind of query which does not give a score, so it is 
+  known as a "non-scoring" query.It is only concerned about answering the question - 
+  "Does this document match?". The answer is always a simple, binary yes or no. This kind of query is said to be made 
+  in a "filtering" context (e.g. Is the created date in the range 2013 - 2014?), hence it is called a filter. Filters are 
+  simple checks for set inclusion or exclusion. The goal of filtering is to reduce the 
```

Comment:
> "The goal of filtering is to reduce the number of documents"

This might be true most of the time, but e.g. `constant_score` isn't really about reducing the document set, its just about saying you don't care about the score. I'd be okay with adding "most of the time" somewhere, maybe also leave out the "instead of..." part, since reducing document set size might also happen in scoring queries.